ADObjectPermissionEntry: (Regression in 6.6.1) Cannot find a provider with the name 'DC=contoso,DC=local,Microsoft.ActiveDirectory.Management.dll\ActiveDirectory'

### Problem description

Using ADObjectPermissionEntry on Windows Server 2025 to grant a service account the permission replicating directory changes, I get this error:
`Cannot find a provider with the name 'DC=contoso,DC=local,Microsoft.ActiveDirectory.Management.dll\ActiveDirectory'. `

### Verbose logs

```text
VERBOSE: [2025-03-17 09:54:12Z] [VERBOSE] [SP]: LCM:  [ Start  Test     ]  
[[ADObjectPermissionEntry]GrantReplicatingDirectoryChanges]
VERBOSE: [2025-03-17 09:54:12Z] [VERBOSE] [SP]:                            
[[ADObjectPermissionEntry]GrantReplicatingDirectoryChanges] Retrieved the AD Drive full PSPath of 'Microsoft.ActiveDirectory.Management.dll\ActiveDirectory:://RootDSE/'. (OPE0008)
VERBOSE: [2025-03-17 09:54:12Z] [VERBOSE] [SP]: LCM:  [ End    Test     ]  
[[ADObjectPermissionEntry]GrantReplicatingDirectoryChanges]  in 0.2260 seconds.
VERBOSE: [2025-03-17 09:54:13Z] [ERROR] PowerShell DSC resource MSFT_ADObjectPermissionEntry  failed to execute Test-TargetResource functionality with error message: Cannot find a provider with the name 'DC=contoso,DC=local,Microsoft.ActiveDirectory.Management.dll\ActiveDirectory'.
```

### DSC configuration

```powershell
ADObjectPermissionEntry GrantReplicatingDirectoryChanges
{
	Ensure                             = 'Present'
	Path                               = "DC=contoso,DC=local"
	IdentityReference                  = "serviceAccountName"
	ActiveDirectoryRights              = 'ExtendedRight'
	AccessControlType                  = 'Allow'
	ObjectType                         = "1131f6aa-9c07-11d1-f79f-00c04fc2dcd2" # Replicate Directory Changes Permission
	ActiveDirectorySecurityInheritance = 'All'
	InheritedObjectType                = '00000000-0000-0000-0000-000000000000'
	PsDscRunAsCredential               = $DomainAdminCredsQualified
}
```

### Suggested solution

It seems to happen because the path contains the domain's DN as a prefix, which was not present in the previous versions of the module:
`DC=contoso,DC=local,`

Show the PSPath value for AD:

```powershell
Import-Module ActiveDirectory
(Get-Item -Path 'AD:').PSPath
Microsoft.ActiveDirectory.Management.dll\ActiveDirectory:://RootDSE/
```

### Operating system the target node is running

```text
OsName               : Microsoft Windows Server 2025 Datacenter Azure Edition
OsOperatingSystemSKU : DatacenterAzureServerEdition
OsArchitecture       : 64-bit
WindowsVersion       : 2009
WindowsBuildLabEx    : 26100.1.amd64fre.ge_release.240331-1435
OsLanguage           : en-US
OsMuiLanguages       : {en-US}
```

### PowerShell version and build the target node is running

```text
PSVersion                      5.1.26100.2161
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.26100.2161
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### ActiveDirectoryDsc version

```text
Name               Version Path
----               ------- ----
ActiveDirectoryDsc 6.6.1   C:\Program Files\WindowsPowerShell\Modules\ActiveDirectoryDsc\ActiveDirectoryDsc.psd1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,11 @@ For older change log history see the [historic changelog](HISTORIC_CHANGELOG.md)
 
 ## [Unreleased]
 
+### Fixed
+ - ADObjectPermissionEntry
+   -Fixed regression in 6.6.1 when using Join-Path.
+   ([issue #727](https://github.com/dsccommunity/ActiveDirectoryDsc/issues/727)).
+
 ## [6.6.1] - 2025-03-15
 
 ### Fixed
@@ -16,7 +21,7 @@ For older change log history see the [historic changelog](HISTORIC_CHANGELOG.md)
     ([issue #650](https://github.com/dsccommunity/ActiveDirectoryDsc/issues/650)).
 - ADObjectPermissionEntry
   - Fixed regression where resource cannot run on Windows Server 2016
-    ([issue #724])(https://github.com/dsccommunity/ActiveDirectoryDsc/issues/724)).
+    ([issue #724](https://github.com/dsccommunity/ActiveDirectoryDsc/issues/724)).
 
 ## [6.6.0] - 2024-09-29
 

--- a/source/DSCResources/MSFT_ADObjectPermissionEntry/MSFT_ADObjectPermissionEntry.psm1
+++ b/source/DSCResources/MSFT_ADObjectPermissionEntry/MSFT_ADObjectPermissionEntry.psm1
@@ -84,7 +84,7 @@ function Get-TargetResource
     try
     {
         # Get the current acl
-        $acl = Get-Acl -Path (Join-Path -Path $ADDrivePSPath -ChildPath $Path) -ErrorAction Stop
+        $acl = Get-Acl -Path ($ADDrivePSPath + $Path) -ErrorAction Stop
     }
     catch [System.Management.Automation.ItemNotFoundException]
     {
@@ -210,7 +210,7 @@ function Set-TargetResource
     $ADDrivePSPath = Get-ADDrivePSPath
 
     # Get the current acl
-    $acl = Get-Acl -Path (Join-Path -Path $ADDrivePSPath -ChildPath $Path)
+    $acl = Get-Acl -Path ($ADDrivePSPath + $Path)
 
     if ($Ensure -eq 'Present')
     {
@@ -255,7 +255,7 @@ function Set-TargetResource
 
     # Set the updated acl to the object
     $acl |
-        Set-Acl -Path (Join-Path -Path $ADDrivePSPath -ChildPath $Path)
+        Set-Acl -Path ($ADDrivePSPath + $Path)
 }
 
 <#

